src/subscription.ts

```diff
@@ -88,6 +89,30 @@ export type SetSubscriptionMetadataResponse = MetadataResponse;
 export type DetachSubscriptionCallback = EmptyCallback;
 export type DetachSubscriptionResponse = EmptyResponse;
 
+// JSDoc won't see these, so this is just to let you get typings
+// in your editor of choice. Callback types are open ended so that
+// not-quite-matching parameters won't cause a breaking change. We
+// may still shift these to required later.
+export declare interface Subscription {
+  on(
+    event: 'message',
+    listener: (() => void) | ((message: Message) => void)
```

Comment:
You could simplify this by collapsing to:  `listener: (message?: Message) => void`

Reply:
Here and below

Reply:
Yeah, I did this in response to @ettancos 's comment. It's sort of misleading to say that `Message` is optional here, because it's not. It feels like a weird hack to get around a TS limitation.

At this point I think I'm prepared to just comment out the typings part for now, and re-add them on the next major. Getting the JSDoc in place is more immediately useful, and I think all of us are nervous about breaking existing clients with typings (for various reasons).